org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseProjectImporter.java

```diff
@@ -43,6 +45,13 @@ public boolean applies(IProgressMonitor monitor) throws CoreException {
 		if (directories == null) {
 			BasicFileDetector eclipseDetector = new BasicFileDetector(rootFolder.toPath(), DESCRIPTION_FILE_NAME)
 					.addExclusions("**/bin");//default Eclipse build dir
+			Preferences preferences = getPreferences();
+			for (IProject project : ProjectUtils.getAllProjects()) {
+				if ((preferences.isImportGradleEnabled() && ProjectUtils.isGradleProject(project)) || (preferences.isImportMavenEnabled() && ProjectUtils.isMavenProject(project))) {
```

Comment:
mmm might be an issue with 3rd party importers (eg. Salesforce's Bazel). Ideally we'd need a generic way to tell is a project is something other than an Eclipse project

Reply:
Fixed.